components/uuf-core/src/test/java/org/wso2/carbon/uuf/internal/filter/CsrfFilterTest.java

```diff
@@ -0,0 +1,171 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.uuf.internal.filter;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.carbon.uuf.api.config.Configuration;
+import org.wso2.carbon.uuf.core.UriPatten;
+import org.wso2.carbon.uuf.internal.auth.SessionRegistry;
+import org.wso2.carbon.uuf.spi.HttpRequest;
+import org.wso2.carbon.uuf.spi.HttpResponse;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test class for {@link CsrfFilter}
+ */
+public class CsrfFilterTest {
+
+    private Configuration configuration;
+    private HttpRequest request;
+    private HttpResponse response;
+    private CsrfFilter csrfFilter;
+
+    @BeforeMethod
+    public void setup() {
+        configuration = mock(Configuration.class);
+        request = mock(HttpRequest.class);
+        response = mock(HttpResponse.class);
+        when(request.getUriWithoutContextPath()).thenReturn("/");
+        csrfFilter = new CsrfFilter();
+    }
+
+    @Test
+    public void doFilterGetRequest() {
+        // Configuration settings
+        when(configuration.getCsrfIgnoreUris()).thenReturn(Collections.emptySet()); // Not csrf
+
+        // Http request settings
+        when(request.isGetRequest()).thenReturn(true); // Get request
+        when(request.getCookieValue(SessionRegistry.CSRF_TOKEN)).thenReturn(null); // No csrf token
+        when(request.getFormParams()).thenReturn(Collections.emptyMap()); // No csrf tokens
+
+        // Assert filter
+        FilterResult result = csrfFilter.doFilter(configuration, request, response);
+        Assert.assertEquals(result.isContinue(), true);
+    }
+
+    @Test
+    public void doFilterPostRequestCsrfInIgnoreList() {
+        // Configuration settings
+        Set<UriPatten> patterns = new HashSet<>();
+        patterns.add(new UriPatten("/"));
+        when(configuration.getCsrfIgnoreUris()).thenReturn(patterns); // Not csrf
+
+        // Http request settings
+        when(request.isGetRequest()).thenReturn(false); // post request
+        when(request.getCookieValue(SessionRegistry.CSRF_TOKEN)).thenReturn(null); // No csrf token
+        when(request.getFormParams()).thenReturn(Collections.emptyMap()); // No csrf tokens
+
+        // Assert filter
+        FilterResult result = csrfFilter.doFilter(configuration, request, response);
+        Assert.assertEquals(result.isContinue(), true);
+    }
+
+    @Test
+    public void doFilterPostRequestCsrfNoToken() {
+        // Configuration settings
+        when(configuration.getCsrfIgnoreUris()).thenReturn(Collections.emptySet()); // Not csrf
```

Comment:
Are we following this pattern of commenting along with the code line? Better if we can split that into two lines.